### PR TITLE
[1.x] Set `APPLICATION_EXTENSION_API_ONLY` to `YES`

### LINKED ISSUE #749: Compilation Errors with Swift 4 and Xcode 9
### What did you do?

Tried to compile my Project with Swift 4 and Xcode 9.

### What did you expect to happen?

That everything will build successfully.

### What actually happened instead?

Quick and Nimble show strange errors. It is kinda strange because I don't get these errors on a different project.

Errors:
screen shot 2017-09-29 at 18 47 06
screen shot 2017-09-29 at 18 48 56

### Environment

List the software versions you're using:

 - Quick: *1.2.0*
 - Nimble: *7.0.2*
 - Xcode Version: Version 9.0 (9A235)
 - Swift Version: `Xcode Default

Please also mention which package manager you used and its version. Delete the
other package managers in this list:

 - Cocoapods: 1.3.1

### PR DESCRIPTION
**Pull request by ikesyo**
This enables us to use Swift internal classes from Objective-C side with importing the generated header.

Fixes #749. See also https://github.com/Quick/Nimble/pull/525.

### CODE REVIEW

**Comment by ikesyo** [comment]
Ping @Quick/core

File: Sources/Quick/DSL/World+DSL.swift [hunk 1]
```diff
@@ -85,6 +85,7 @@ extension World {
     }
 #endif
 
+    @nonobjc
```
Comment (sharplet) [inline Sources/Quick/DSL/World+DSL.swift]: Is this necessary? Does `World` use `@objcMembers`?

Comment (ikesyo) [reply Sources/Quick/DSL/World+DSL.swift]: Yes it's needed. Otherwise those conflict with `objc_it` and so on.

**Comment by ikesyo** [comment]
I'll consider this as well: https://github.com/Quick/Nimble/issues/540

**Comment by ikesyo** [comment]
Thanks @sharplet!

**Pull request merged**